### DISCUSSION TITLE
Save scans to a pdf file locally (even when paperless instance is unavailable)

### DISCUSSION THREAD

**Discussion by klys-equinix** [astubenbord/paperless-mobile#174] [Feature Requests]
I would like to be able to use the app as a simple scanner when i do not have access to paperless. A simple button on the scanning page saying 'Export to PDF' next to upload button would be nice. I would be interested in developing this feature, please provide info which branch i should start from

**Comment by astubenbord** [comment]
Hi, why don't you just use a standalone document scanner for this? They yield scans with much better quality than the in-app scanner. I think it would make more sense to add a feature such that if you're offline, the request is added to a queue and once the device is connected to the internet again, the files are automatically consumed from the queue and uploaded to the server.

**Reply by klys-equinix** [reply]
Mostly to not have to keep more apps on my phone. But the queue functionality is more fitting for this app, i agree

**Reply by klys-equinix** [reply]
Also, any plans on improving the scanner?

**Reply by klys-equinix** [reply]
Thirdly, i did not find any FOSS app with a better scanner

**Reply by astubenbord** [reply]
> Also, any plans on improving the scanner? 

Yes, there are! I've already started to work on it but the flutter camera package is pretty limited for streaming. But I found that camera_awesome seems to have improved quite a bit, so I'd like to look at that. If you're interested in collaboration just let me know.

**Reply by klys-equinix** [reply]
Yes, I will be happy to help. You can tell me what to do. I probably won't have tons of time tough

**Reply by astubenbord** [reply]
Same here sadly, but it's an enhancement so it's not urgent at all. We can discuss the details inside the [Matrix Chatroom](https://matrix.to/#/!CwmZwPHEyikzpxXPxl:matrix.org?via=matrix.org) if you like.

**Reply by klys-equinix** [reply]
I think you would need to invite me (matrix login losiu97)

**Reply by klys-equinix** [reply]
since i get 403

**Reply by astubenbord** [reply]
I invited you now
